### ISSUE TITLE
estimated training time

### ISSUE THREAD

**Issue by Yipinggggg** [isyangshu/Surgformer#2]
Hi, thanks for the nice work!
I tried to implement your code but found that the training was very slow. I saw that you use distributed training in the code. Could you kindly provide more info on your training setting? i.e. GPU, number of GPUs, and how much time is needed for convergence?

**Comment by isyangshu** [comment]
You'd better pre-process the images to get low-resolution frames first, otherwise you will wait on IO. You can check your data time to determine if the bottleneck is IO.

I use 2~4 GPUs (3090) for training, and uses path of pre-processed frames (short edge: 256).

It will take about 10~20 epochs for the best performance.

I spend about 2 days for each dataset, i.e., Cholec80 and AutoLaparo. Each epoch will spend about 1.30 hours for Cholec80 with 2 GPUs.

**Comment by Yipinggggg** [comment]
Thanks for the reply! On a different note, will enable_deepspeed affect results largely? I scrolled the code and noticed that there are some loss scalers used specifically for deepspeed. Are the results still reproducible without deepspeed? Thanks!

**Comment by isyangshu** [comment]
Generally speaking the results are approximate, I did not specifically experiment.

**Comment by ganshuwei** [comment]
> You'd better pre-process the images to get low-resolution frames first, otherwise you will wait on IO. You can check your data time to determine if the bottleneck is IO.
> 
> I use 2~4 GPUs (3090) for training, and uses path of pre-processed frames (short edge: 256).
> 
> It will take about 10~20 epochs for the best performance.
> 
> I spend about 2 days for each dataset, i.e., Cholec80 and AutoLaparo. Each epoch will spend about 1.30 hours for Cholec80 with 2 GPUs.

Could I ask how many epochs did you train for each dataset in two days? 20? Is the speed difference significant without using DeepSpeed?
